client-src/css/shared-css.js

```diff
@@ -134,7 +134,11 @@ export const SHARED_STYLES = [
   }
 
   .preformatted {
-    white-space: pre-wrap;
+    overflow-wrap: anywhere;
```

Comment:
The only purpose of `.preformatted` is to use `pre-wrap`.  If you want some text to not wrap, it would be better to remove the `class="reformatted"` from that text.

I think this part of the requirements is a little tricky.  We do want users to be able to enter text that has extra newline characters in it and then have that render as paragraphs.